iroha/src/sumeragi/mod.rs

```diff
@@ -94,7 +99,8 @@ where
     events_sender: EventsSender,
     wsv: Arc<WorldStateView<W>>,
     /// Hashes of the transactions that were forwarded to a leader, but not yet confirmed with a receipt.
-    transactions_awaiting_receipts: Arc<DashSet<Hash>>,
+    /// And time at which this transaction was sent to the leader by this peer.
+    transactions_awaiting_receipts: Arc<DashMap<Hash, Duration>>,
```

Comment:
Also could you rename it to `tx_awaiting_receipts`?

Reply:
`txs_awaiting_receipts` then I think